src/input/findTSLintConfiguration.ts

```diff
@@ -1,44 +1,67 @@
+import { findRawConfiguration } from "./findRawConfiguration";
+import { findReportedConfiguration } from "./findReportedConfiguration";
 import { Exec } from "../adapters/exec";
-import { findConfiguration } from "./findConfiguration";
+import { OriginalConfigurations } from "./findOriginalConfigurations";
+import { SansDependencies } from "../binding";
+import { importer } from "./importer";
 
 export type TSLintConfiguration = {
+    extends?: string[];
     rulesDirectory: string[];
     rules: TSLintConfigurationRules;
 };
 
-export type TSLintConfigurationRules = {
-    [i: string]: any;
-};
+export type TSLintConfigurationRules = Record<string, any>;
 
 const defaultTSLintConfiguration = {
+    extends: [],
     rulesDirectory: [],
     rules: {},
 };
 
 export type FindTSLintConfigurationDependencies = {
     exec: Exec;
+    importer: SansDependencies<typeof importer>;
 };
 
 export const findTSLintConfiguration = async (
     dependencies: FindTSLintConfigurationDependencies,
     config: string | undefined,
-): Promise<TSLintConfiguration | Error> => {
-    const rawConfiguration = await findConfiguration<TSLintConfiguration>(
-        dependencies.exec,
-        "tslint --print-config",
-        config || "./tslint.json",
-    );
+): Promise<OriginalConfigurations<TSLintConfiguration> | Error> => {
+    const filePath = config || "./tslint.json";
+    const [rawConfiguration, reportedConfiguration] = await Promise.all([
+        findRawConfiguration<Partial<TSLintConfiguration>>(dependencies.importer, filePath, {
+            extends: [],
+        }),
+        findReportedConfiguration<TSLintConfiguration>(
+            dependencies.exec,
+            "tslint --print-config",
+            config || "./tslint.json",
+        ),
+    ]);
 
-    if (rawConfiguration instanceof Error) {
-        if (rawConfiguration.message.includes("unknown option `--print-config")) {
+    if (reportedConfiguration instanceof Error) {
+        if (reportedConfiguration.message.includes("unknown option `--print-config")) {
             return new Error("TSLint v5.18 required. Please update your version.");
         }
 
+        return reportedConfiguration;
+    }
+
+    if (rawConfiguration instanceof Error) {
         return rawConfiguration;
     }
 
     return {
-        ...defaultTSLintConfiguration,
-        ...rawConfiguration,
+        full: {
+            ...defaultTSLintConfiguration,
+            ...rawConfiguration,
```

Comment:
Hi @JoshuaKGoldberg I noticed TSLint rule arguments are not being picked up in the latest release and I think the issue might be here. Should this be `reportedConfiguration` instead?

Reply:
Perhaps! @jeremyyap would you mind filing an issue! I can take a look later today.